### ISSUE TITLE
Use 'geotorch.orthogonal' (and similarly 'torch.nn.utils.parametrizations.orthogonal') on a linear layer but got 'grad = None' of the weight of the layer

### ISSUE THREAD

**Issue by Jusker-Greek** [lezcano/geotorch#44]
when i use the tool 'geotorch.orthogonal' in my code, the debug show that 'linear.weight' will become 'linear.parametrization' and 'linear.parametrization.grad=None'
I wonder if it's normal and whether this will affect the gradient and optimize of the linear layer parmeter?

i supposed it was the usage of 'geotorch' caused the non grad case, because when i don't include that line the layer have grad

**Comment by lezcano** [comment]
After parametrising `weight`, `weight` is no longer a leaf in your graph, so it won't have a grad, but it will have a backward function. The new parameters are now called `linear.parametrizations.weight.original`, which does have a gradient.

All this is discussed at https://pytorch.org/tutorials/intermediate/parametrizations.html#inspecting-a-parametrized-module

**Comment by Jusker-Greek** [comment]
thanks for your fast reply!
So the weight of the layer still be optimized properly?
(I got to think this question because a use this single linear layer as the whole neural network and i observed that the optimization seem to be not very normal and loss didn't decrease, so i supposed that it was parametrization that causes this)
thanks a lot

**Comment by lezcano** [comment]
No problem! Closing
